src/components/ui/buttons/ClaimButton.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <button class="claim-btn" v-if="reward" @click.stop="actionHandler">
+    <img
+      class="claim-btn-icon"
+      src="@/assets/images/deposit.svg"
+      alt="Deposit"
+    />
+    Claim
+  </button>
+</template>
+
+<script>
+import { mapGetters } from "vuex";
+import { getCvxClaimableReward } from "@/helpers/cauldron/getCvxClaimableReward.ts";
+import { handleClaimCrvReward } from "@/helpers/cauldron/handleClaimCrvReward.ts";
+export default {
+  props: {
+    cauldron: { type: Object, require: true },
+  },
+
+  data() {
+    return {
+      reward: null,
+    };
+  },
+
+  computed: {
+    ...mapGetters({
+      account: "getAccount",
+    }),
+  },
+
+  watch: {
+    async account() {
+      await this.getReward();
+    },
+  },
+
+  methods: {
+    async actionHandler() {
+      const { collateral } = this.cauldron.contracts;
+      await handleClaimCrvReward(collateral, this.account);
+    },
+
+    async getReward() {
+      const { collateral } = this.cauldron.contracts;
+      const { decimals } = this.cauldron.config.collateralInfo;
```

Comment:
Component is called ClaimButton but there's there's specific calls to `getCvxClaimableReward` / `handlerClaimCrvReward` here. Should be a component parameter so ClaimButton stands to its generalized aspect.